client/verta/verta/pipeline/_pipeline_step.py

```diff
@@ -0,0 +1,213 @@
+# -*- coding: utf-8 -*-
+
+from typing import Any, Dict, List, Optional
+
+from verta._internal_utils._utils import Configuration, Connection
+from verta.registry.entities import RegisteredModel, RegisteredModelVersion
+
+
+class PipelineStep:
+    """
+    A single step within an inference pipeline, representing a single model version to be run.
+
+    Parameters
+    ----------
+    name : str
+        Name of the step, for use within the scope of the pipeline only.
+    model_version : :class:`~verta.registry.entities.RegisteredModelVersion`
+        Registered model version to run for the step.
+    predecessors : list, optional
+        List of PipelineSteps whose outputs will be treated as inputs to this step.
+        If not included, the step is assumed to be an initial step. Values must be unique.
+
+    Attributes
+    ----------
+        name : str
+            Name of the step within the scope of the pipeline.
+        model_version
+            :class:`~verta.registry.entities.RegisteredModelVersion` run by this step.
+        predecessors : list
+            List of PipelineSteps whose outputs will be treated as inputs to this step.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        model_version: RegisteredModelVersion,
+        predecessors: Optional[
+            List["PipelineStep"]
+        ] = None,  # Optional because it could be the first step with no predecessors
+    ):
+        self._name = name
+        self._model_version = model_version
+        self._predecessors = predecessors or list()
+
+        # avoid the need to pass in connection params when building as local object
+        self._registered_model: Optional[RegisteredModel] = None
+
+        # avoid additional http calls to refresh RMV cache.
+        self._registered_model_id = self._model_version.registered_model_id
+
+    def __repr__(self) -> str:
+        sequence = (
+            "\nPipelineStep:",
+            f"step name: {self.name}",
+        )
+        if self._registered_model:  # don't display on local-only objects`
+            sequence += (f"registered_model: {self._registered_model.name}",)
```

Comment:
Because an RMV has an attribute for the _id_ of then related RM, but not for the _name_ of that RM, we have to fetch the name to display it here, which requires client connection params (conn and conf).

When a user is declaring a pipeline step object locally (before registering the pipeline for the first time), they presumably already know the RM, and RMV they are using because they explicitly fetch both and then pass in the RMV as a param.  We don't want to require them to pass in connection params as well at that point.

However, when a user fetches an existing registered pipeline, they don't have that context for each step's RM/RMV because they didn't just assemble it themself.  In that scenario, just having the RMV name (like "v1" for example, which is fairly ubiquitous in our own examples) begs the question "v1 of **what exactly**?"  We could display only the RM's _id_, but that requires a bunch of extra steps on the user's part just to figure out which actual model they are dealing with (Digging in the UI, or making extra API calls)

This way, users pulling down an existing pipe get the immediate context they need on each step (which model and which version of it), and users building a new pipeline (locally only, to start) don't have to insert connection params to get context they already have.